src/frontend/src/lib/components/qr/VipQrCodeModal.svelte

```diff
@@ -0,0 +1,137 @@
+<script lang="ts">
+	import { Modal, QRCode } from '@dfinity/gix-components';
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { onDestroy, onMount } from 'svelte';
+	import IconAstronautHelmet from '$lib/components/icons/IconAstronautHelmet.svelte';
+	import ReceiveCopy from '$lib/components/receive/ReceiveCopy.svelte';
+	import Button from '$lib/components/ui/Button.svelte';
+	import ButtonCloseModal from '$lib/components/ui/ButtonCloseModal.svelte';
+	import ButtonGroup from '$lib/components/ui/ButtonGroup.svelte';
+	import ContentWithToolbar from '$lib/components/ui/ContentWithToolbar.svelte';
+	import SkeletonText from '$lib/components/ui/SkeletonText.svelte';
+	import { VIP_CODE_REGENERATE_INTERVAL } from '$lib/constants/app.constants';
+	import {
+		VIP_CODE_REGENERATE_BUTTON,
+		VIP_QR_CODE_COPY_BUTTON
+	} from '$lib/constants/test-ids.constants';
+	import { authIdentity } from '$lib/derived/auth.derived';
+	import { nullishSignOut } from '$lib/services/auth.services';
+	import { getNewReward } from '$lib/services/reward-code.services';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { modalStore } from '$lib/stores/modal.store';
+	import { replacePlaceholders } from '$lib/utils/i18n.utils';
+
+	let counter = VIP_CODE_REGENERATE_INTERVAL;
+	let countdown: NodeJS.Timeout | undefined;
+	const maxRetries = 3;
+	let retries = 0;
```

Comment:
For better readability, could you either add a comment indicating that this is for retries in case of errors or rename the variable to make it more explicit?